rust/tests/vacuum_test.rs

```diff
@@ -76,3 +85,261 @@ async fn vacuum_delta_8_0_table() {
+        .create_table_from_schema(get_xy_date_schema(), &[])
+        .await;
+    let clock = TestClock::from_systemtime();
+
+    let paths = ["delete_me.parquet", "dont_delete_me.parquet"];
+
+    for path in paths {
+        context
+            .add_file(
+                path,
+                "random junk".as_ref(),
+                &[],
+                clock.current_timestamp_millis(),
+                true,
+            )
+            .await;
+    }
+
+    clock.tick(Duration::seconds(10));
+
+    context
+        .remove_file("delete_me.parquet", &[], clock.current_timestamp_millis())
+        .await;
+
+    let res = {
+        clock.tick(Duration::days(8));
+        let table = context.table.as_mut().unwrap();
+        let mut plan = Vacuum::default();
+        plan.clock = Some(Arc::new(clock.clone()));
+        plan.execute(table).await.unwrap()
+    };
+
+    assert_eq!(res.files_deleted.len(), 1);
+    assert!(is_deleted(&mut context, "delete_me.parquet").await);
+    assert!(!is_deleted(&mut context, "dont_delete_me.parquet").await);
+}
+
+#[tokio::test]
+// Validate vacuum works on a table with multiple partitions
+async fn test_partitioned_table() {
+    let mut context = TestContext::from_env().await;
+    context
+        .create_table_from_schema(get_xy_date_schema(), &["date", "x"])
+        .await;
+    let clock = TestClock::from_systemtime();
+
+    let paths = [
+        "date=2022-07-03/x=2/delete_me.parquet",
+        "date=2022-07-03/x=2/dont_delete_me.parquet",
+    ];
+    let partition_values = [("date", Some("2022-07-03")), ("x", Some("2"))];
+
+    for path in paths {
+        context
+            .add_file(
+                path,
+                "random junk".as_ref(),
+                &partition_values,
+                clock.current_timestamp_millis(),
+                true,
+            )
+            .await;
+    }
+
+    clock.tick(Duration::seconds(10));
+
+    context
+        .remove_file(
+            "date=2022-07-03/x=2/delete_me.parquet",
+            &partition_values,
+            clock.current_timestamp_millis(),
+        )
+        .await;
+
+    let res = {
+        clock.tick(Duration::days(8));
+        let table = context.table.as_mut().unwrap();
+        let mut plan = Vacuum::default();
+        plan.clock = Some(Arc::new(clock.clone()));
+        plan.execute(table).await.unwrap()
+    };
+
+    assert_eq!(res.files_deleted.len(), 1);
+    assert!(is_deleted(&mut context, "date=2022-07-03/x=2/delete_me.parquet").await);
+    assert!(!is_deleted(&mut context, "date=2022-07-03/x=2/dont_delete_me.parquet").await);
+}
+
+#[tokio::test]
+// Validate that files and directories that start with '.' or '_' are ignored
+async fn test_ignored_files() {
+    let mut context = TestContext::from_env().await;
+    context
+        .create_table_from_schema(get_xy_date_schema(), &["date"])
+        .await;
+    let clock = TestClock::from_systemtime();
+
+    let paths = [
+        ".dotfile",
```

Comment:
The official documentation make a call out on ignoring directories that start with a underscore and [the reference implementation's tests](https://github.com/delta-io/delta/blob/6707b29f58a30989f40478ce40b3200c48bf37ce/core/src/test/scala/org/apache/spark/sql/delta/DeltaVacuumSuite.scala#L130) also includes invisible files (dot files) which aren't documented

These are unmanaged files so I can merge it with that test if you think it would help clarify things.

Reply:
Yeah I was confused because you were calling `add_file` on them and that adds them to the Delta log. In their test, they set `commitToActionLog => false` for these. So yeah I think moving to unmanaged files test would be better.